### PR TITLE
fix(Core/Spell): Add aditional check to LoadSpellScripts()

### PR DESCRIPTION
**Pull request by Kitzunu**
## Changes Proposed:
- Found in TC. Good check =)

## Tests Performed:

- None

## How to Test the Changes:

1. Try to use effect index >= 3 in db

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Globals/ObjectMgr.cpp [hunk 1]
```diff
@@ -5132,10 +5132,15 @@ void ObjectMgr::LoadSpellScripts()
             continue;
         }
 
-        uint8 i = (uint8)((uint32(itr->first) >> 24) & 0x000000FF);
+        SpellEffIndex i = SpellEffIndex((uint32(itr->first) >> 24) & 0x000000FF);
+        if (uint32(i) >= MAX_SPELL_EFFECTS)
+        {
+            LOG_ERROR("sql.sql", "Table `spell_scripts` has too high effect index %u for spell (Id: %u) as script id", uint32(i), spellId)
```
Comment (Kitzunu) [inline src/server/game/Globals/ObjectMgr.cpp:5138]:
```suggestion
            LOG_ERROR("sql.sql", "Table `spell_scripts` has too high effect index %u for spell (Id: %u) as script id", uint32(i), spellId);
```

**Comment by FrancescoBorzi** [comment]
@Kitzunu if you can just run the core and access the game, we can consider this as tested

**Comment by Kitzunu** [comment]
Will do

**Comment by stoudtlr** [comment]
> @Kitzunu if you can just run the core and access the game, we can consider this as tested

built server with PR and was able to enter world
image

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by Kitzunu** [comment]
There is no merge conflicts? @FrancescoBorzi

**Review by Helias — APPROVED** [review]

**Pull request merged**